lib/katello/tasks/upgrades/3.15/reindex_rpm_modular.rake

```diff
@@ -0,0 +1,36 @@
+namespace :katello do
+  namespace :upgrades do
+    namespace '3.15' do
+      desc "Reindex the is_modular attribute of an rpm"
+      def update_modular(pulp_response)
+        modular = []
+        non_modular = []
+        pulp_response.each do |rpm|
+          if rpm["is_modular"]
+            modular << rpm["_id"]
+          else
+            non_modular << rpm["_id"] unless rpm["is_modular"]
+          end
+        end
+        ::Katello::Rpm.where(pulp_id: modular).update_all(modular: true)
+        ::Katello::Rpm.where(pulp_id: non_modular).update_all(modular: false)
+      end
+
+      task :reindex_rpm_modular, [:input_file] => ["environment"] do
+        User.current = User.anonymous_api_admin
+        criteria = { :fields => ["is_modular"],
+                     :limit => SETTINGS[:katello][:pulp][:bulk_load_size],
+                     :skip => 0
```

Comment:
i think it might be worth it to limit to ONLY those rpms that have is_modular set to true.  That way we'd likely cut down the migration time by like ~90%, or even more depending on if the user has rhel 8 or centos 8 synced

Reply:
> i think it might be worth it to limit to ONLY those rpms that have is_modular set to true. That way we'd likely cut down the migration time by like ~90%, or even more depending on if the user has rhel 8 or centos 8 synced

updated